Add Illustration Component

qyiming opened the pull request:
Fixes #1360

### Additional description
Add a new DSR illustration component per the spec here: https://lightningdesignsystem.com/components/illustration/

@interactivellama please review.
---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

qyiming commented:
@interactivellama please review.

fyi @garygong @davidlygagnon

interactivellama commented:
@qyiming Lint isn't passing. It looks like it's a style issue that `run lint:fix` should fix but I'm not sure.

qyiming commented:
@interactivellama Let me double check, though the last I commit I pushed for this PR contained code atomically updated by npm run lint:fix

qyiming commented:
@interactivellama please review again. I ran npm run lint:fix on my branch again just now, and it fixed one more issue this time. All checks have passed now for this PR.

qyiming commented on components/illustration/index.jsx:
```diff
@@ -0,0 +1,149 @@
+				className="slds-illustration__svg"
+				aria-hidden="true"
+				data={illustration}
+				name={kababCaseName}
```
I was following the example in from components/utilities/utility-icon/index.jsx to include "name" in the <Svg>

qyiming commented on assets/images/illustrations/NoAccess.svg:
```diff
@@ -0,0 +1,49 @@
+<svg width="300px" height="200px" viewBox="0 0 300 200" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" id="no-access">
```
not in this release, probably in the next one.

qyiming commented:
@interactivellama Thanks for the review, I have addressed your comments and the doc site is generated correctly. Please review again.

<img width="1436" alt="screen shot 2018-07-11 at 2 50 36 pm" src="https://user-images.githubusercontent.com/4957585/42601303-0c5db4e4-851a-11e8-90ca-6a18d41e3842.png">

One other thing I noticed is that we probably need to add a illustration image to public/static/images/component-examples/ in design-system-react-sit repo, let me know if you'd like me to add one.

interactivellama commented:
This isn't something to be fixed, but since we are adding the 400px height. This can happen.
![screen shot 2018-07-11 at 11 03 29 pm](https://user-images.githubusercontent.com/1290832/42613549-c22cd6ee-855e-11e8-9d4c-eb8ee392ac5f.png)
